fv3core/fv3core/stencils/c_sw.py

```diff
@@ -416,22 +423,31 @@ def __init__(
         )
         origin_halo1 = (grid_indexing.isc - 1, grid_indexing.jsc - 1, 0)
         self.delpc = utils.make_storage_from_shape(
-            grid_indexing.max_shape, origin=origin_halo1
+            grid_indexing.max_shape,
+            origin=origin_halo1,
+            backend=stencil_factory.backend,
         )
         self.ptc = utils.make_storage_from_shape(
-            grid_indexing.max_shape, origin=origin_halo1
+            grid_indexing.max_shape,
+            origin=origin_halo1,
+            backend=stencil_factory.backend,
         )
         self._initialize_delpc_ptc = stencil_factory.from_dims_halo(
             initialize_delpc_ptc,
             compute_dims=[X_DIM, Y_DIM, Z_DIM],
             compute_halos=(3, 3),
         )
 
-        self._tmp_ke = utils.make_storage_from_shape(grid_indexing.max_shape)
-        self._tmp_vort = utils.make_storage_from_shape(grid_indexing.max_shape)
-        self._tmp_fx = utils.make_storage_from_shape(grid_indexing.max_shape)
-        self._tmp_fx1 = utils.make_storage_from_shape(grid_indexing.max_shape)
-        self._tmp_fx2 = utils.make_storage_from_shape(grid_indexing.max_shape)
+        def make_storage():
+            return utils.make_storage_from_shape(
+                grid_indexing.max_shape, backend=stencil_factory.backend
+            )
+
+        self._tmp_ke = make_storage()
+        self._tmp_vort = make_storage()
+        self._tmp_fx = make_storage()
+        self._tmp_fx1 = make_storage()
+        self._tmp_fx2 = make_storage()
```

Comment:
Curious to see what others think...

Reply:
I don't know if there's one standard that's best for every module - it depends a lot on how many allocations there are, how many of them use exactly the same arguments (as these do), how many use different args and which ones are different, whether `origin` is being set, whether `backend` already exists as its own variable in the init, how long the variable names being set are, etc. etc. . In some modules, I did go the explicit argument route, but when it would multiply a large number of allocation lines by 3, I opted for this pattern.

The balance would also be different if e.g. we renamed make_storage_from_shape to make_storage.